File: frontend/src/services/favoriteService.js
import API from './api';

export const favoriteService = {
  addFavorite: async (propertyId) => {
    const response = await API.post(`/api/favorites/${propertyId}`);
    return response.data;
  },

  removeFavorite: async (propertyId) => {
    const response = await API.delete(`/api/favorites/${propertyId}`);
    return response.data;
  },

  getFavorites: async () => {
    const response = await API.get('/api/favorites');
    return response.data;
  },

  isFavorite: async (propertyId) => {
    const response = await API.get(`/api/favorites/${propertyId}/status`);
    return response.data;
  }
};

export default favoriteService;

File: backend/src/main/java/com/realestate/service/FavoriteService.java
package com.realestate.service;

import com.realestate.dto.response.PropertyResponse;
import java.util.List;

public interface FavoriteService {
    void addFavorite(Long propertyId);
    void removeFavorite(Long propertyId);
    List<PropertyResponse> getFavorites();
    Boolean isFavorite(Long propertyId);
}
